lib/backend/k8s/syncer.go

```diff
@@ -506,9 +506,11 @@ func (syn *kubeSyncer) readFromKubernetesAPI() {
 				continue
 			}
 			// Event is OK - parse it and send it over the channel.
-			kvp := syn.parseNetworkPolicyEvent(event)
-			latestVersions.networkPolicyVersion = kvp.Revision.(string)
-			syn.sendUpdates([]model.KVPair{*kvp}, KEY_NP)
+			kvp, err := syn.parseNetworkPolicyEvent(event)
+			if err == nil {
+				latestVersions.networkPolicyVersion = kvp.Revision.(string)
+				syn.sendUpdates([]model.KVPair{*kvp}, KEY_NP)
```

Comment:
Do you need these changes anymore?

Feels like whilst they are good we should at least handle the error case by treating as a delete.

Reply:
Yeah, we don't strictly need this change.

I think treating as a delete is going to be more work than it is worth given we don't have the key available here.

Reply:
@robbrockbank I adjusted so that so long as a KVP is returned, it will be treated as a delete.

Note that as the code is written today, this branch will never get hit because err will always be nil. This will at least work in the future if an error _is_ ever returned from this func.